src/service/matchers/lineage_matcher.py

```diff
@@ -3,39 +3,46 @@
 """
 
 import logging
+import re
+from typing import Any, Self
 
-from pydantic import ConfigDict, BaseModel, Field
-from typing import Any
+from pydantic import ConfigDict, BaseModel, Field, model_validator
 
 from src.common.constants import GTDB_UNCLASSIFIED_PREFIX
 from src.common.gtdb_lineage import GTDBRank, GTDBLineage, GTDBLineageError
 from src.common.storage.collection_and_field_names import FLD_GENOME_ATTRIBS_GTDB_LINEAGE
 from src.service import errors
 from src.service import models
 from src.service.app_state_data_structures import PickleableDependencies
-from src.service.processing import CollectionProcess, Heartbeat, HEARTBEAT_INTERVAL_SEC
 from src.service.data_products import genome_attributes
 from src.service.matchers.common_models import Matcher
-
+from src.service.processing import CollectionProcess, Heartbeat, HEARTBEAT_INTERVAL_SEC
 
 # See the workspace specs for the types listed in the matcher
 _GTDB_LINEAGE_METADATA_KEY = "GTDB_lineage"
 _GTDB_LINEAGE_VERSION_METADATA_KEY = "GTDB_source_ver"
 
-_COLLECTION_PARAMS_GTDB_VERSION_KEY = "gtdb_version"
+_COLLECTION_PARAMS_GTDB_VERSIONS_KEY = "gtdb_versions"
 
 
 class GTDBLineageMatcherCollectionParameters(BaseModel):
     "Parameters for the GTDB lineage matcher."
-    gtdb_version: str = Field(
-        example="207.0",
-        description="The GTDB version of the collection in which the matcher is installed. " +
-            "Input data to the matcher must match this version of GTDB or the match will " +
+    gtdb_versions: list[str] = Field(
```

Comment:
I was thinking we could allow just a single string as well to allow for backwards compatibility... but maybe that's more trouble than it's worth

Reply:
And it would mean we don't have to redo the old collections and the old versions will still work... eh, maybe it is worth doing. We could keep the old field and document it as deprecated. Only one field should have a non-null value. What do you think?

Reply:
if it's just for backwards compatibility, I don' t think it's worth it. There have been numerous recent changes (samples flattening, WS genome meta) that don't support backward compatibility.

Reply:
True, but this particular change will break everything since the collection docs themselves won't pass validation once this change goes live. We won't even be able to pull the old collections. But fair enough